src/main/java/org/sonarqube/gradle/SonarQubePlugin.java

```diff
@@ -339,14 +338,14 @@ private void configureJdkSourceAndTarget(Project project, Map<String, Object> pr
   }
 
   private static String getProjectKey(Project project) {
-    // SonarQube uses project keys in URL parameters without internally URL-encoding them.
-    // According to my manual tests with sonar-gradle plugin based on Sonar Runner 2.0 and Sonar 3.4.1,
-    // the current defaults will only cause a problem if project.group or project.name of
-    // the Gradle project to which the plugin is applied contains special characters.
-    // (':' works, ' ' doesn't.) In such a case, sonar.projectKey can be overridden manually.
-    String name = project.getName();
-    String group = project.getGroup().toString();
-    return group.isEmpty() ? name : (group + ":" + name);
+    Project rootProject = project.getRootProject();
+    String rootProjectName = rootProject.getName();
+    String rootGroup = rootProject.getGroup().toString();
+    String rootKey = rootGroup.isEmpty() ? rootProjectName : (rootGroup + ":" + rootProjectName);
+    if (project == rootProject) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)